main.js

```diff
@@ -146,3 +80,167 @@ function stateChangeRow(taskIndex,newRow) {
     }
   }
 }
+// タスク進捗別表示機能
+radioButtons.forEach(radioButton => {
+  radioButton.addEventListener('change', event => {
+    const checkedRadioButton = event.target;
+    console.log('チェックされたラジオボタンは' + radioButton);
+
+    if (checkedRadioButton.checked) {
+      const selectedValue = checkedRadioButton.value;
+      console.log('選択された値:', selectedValue);
+      tasksDisplay(selectedValue);
+    }
+  });
+});
+
+function tasksDisplay(selectedValue) {
+
+  if (selectedValue === 'all') {
+    const targetValue = selectedValue;
+    console.log(targetValue);
+
+    const filteredTasks = tasks;
+    if (filteredTasks) {
+      for (const task of tasks) {
+        console.log('この節内にはいる');
+        filteredTasks.forEach((filteredTask) => {
+          const taskIndex = tasks.findIndex(task => task.id === filteredTask.id);
+          filteredTask.taskIndex = taskIndex + 1;
+        });
+        if (filteredTasks !== null) {
+          displayChange(filteredTasks, selectedValue);
+        }
+
+      }
+    }
+
+  }
+  else if (selectedValue === 'working') {
+    const targetValue = selectedValue;
+    console.log(targetValue);
+    const filteredTasks = tasks.filter(task => task.taskStatus === "working");
+    filteredTasks.forEach((filteredTask) => {
+      const taskIndex = tasks.findIndex(task => task.id === filteredTask.id);
+      filteredTask.taskIndex = taskIndex + 1;
+    });
+    if (filteredTasks !== null) {
+      displayChange(filteredTasks, selectedValue);
+    }
+  }
+  else if (selectedValue === 'completed') {
+    const targetValue = selectedValue;
+    console.log(targetValue);
+    const filteredTasks = tasks.filter(task => task.taskStatus === "completed");
+    filteredTasks.forEach((filteredTask) => {
+      const taskIndex = tasks.findIndex(task => task.id === filteredTask.id);
+      filteredTask.taskIndex = taskIndex + 1;
+    });
+    if (filteredTasks !== null) {
+      displayChange(filteredTasks, selectedValue);
+    }
+    console.log(filteredTasks);
+  }
+
+}
+
+function displayChange(filteredTasks, selectedValue) {
+  if (tBody) {
+    // 子要素をすべて削除する
+    while (tBody.firstChild) {
+      tBody.firstChild.remove();
+    }
+  }
+  if (filteredTasks === 'all') {
+
+  }
```

Comment:
このif文の分岐に入ったときに何も処理をしないのであれば、
```suggestion
  if (filteredTasks !== 'all') {

  }
```
という形にして、中に現在のelse文の処理が書けます。